bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java

```diff
@@ -80,68 +85,137 @@ public static void main(String args[]) throws IOException, KeyStoreException,
+                                            }
+                                        });
+                            }
+                            csvSchema = csvSchemaBuilder.build().withHeader();
+
+                            // Write the user data to csv file.
+                            csvMapper.writerFor(ArrayNode.class)
+                                    .with(csvSchema)
+                                    .writeValue(new FileWriter(file), usersArrayNode);
+
+                            // Remove the header from the CSV schema
+                            csvSchema = csvSchemaBuilder.build();
+
+                            isSchemaInitialized = true;
+                        } else {
+                            // Append the user data to csv file.
+                            csvMapper.writerFor(ArrayNode.class)
+                                    .with(csvSchema)
+                                    .writeValue(new FileWriter(file, true), usersArrayNode);
+                        }
+
+                        if (arrayNode.size() < batchCount) {
+                            LOGGER.log(Level.INFO, "End of results reached.");
+                            break;
+                        }
+                    } else {
+                        LOGGER.log(Level.INFO, "End of results reached.");
+                        break;
                     }
-                }
 
-                // Create a csv schema and generate the csv file containing user information.
-                CsvSchema.Builder csvSchemaBuilder = CsvSchema.builder();
-                for (int i = 0; i < usersArrayNode.size(); i++) {
-                    usersArrayNode.get(i).fieldNames().forEachRemaining(
-                            fieldName -> {
-                                if (!csvSchemaBuilder.hasColumn(fieldName)) {
-                                    csvSchemaBuilder.addColumn(fieldName);
-                                }
-                            });
+                } else {
+                    LOGGER.log(Level.SEVERE, request.getMethod() + " Request to " + request.getURI().toString() +
+                            " returned the status code : " + response.getStatusLine());
+                    return;
                 }
-                CsvSchema csvSchema = csvSchemaBuilder.build().withHeader();
-
-                CsvMapper csvMapper = new CsvMapper();
-                csvMapper.writerFor(ArrayNode.class)
-                        .with(csvSchema)
-                        .writeValue(new File(csvDirectory), usersArrayNode);
-                LOGGER.log(Level.INFO, "User information was successfully written to : " + csvDirectory + " file.");
-            } else {
-                LOGGER.log(Level.INFO, request.getMethod() + " Request to " + request.getURI().toString() +
-                        " returned the status code : " + response.getStatusLine());
             }
         }
+
+        if (isSchemaInitialized) {
```

Comment:
Can't this be true even if the response is empty? https://github.com/wso2/samples-is/pull/285/files#diff-3bdf817a7f1126e2503d504fffe6443192cf741403726665457ef079d6f65ed9R190

Reply:
No, [`usersNode.isArray()`](https://github.com/wso2/samples-is/pull/285/files#diff-3bdf817a7f1126e2503d504fffe6443192cf741403726665457ef079d6f65ed9R162) itself will return false, if the response is empty.